### PR TITLE
Fixes re-ordering of layers

### LINKED ISSUE #4408: Layer reordering breaks if done twice
layer_order Layer_name

EXPECTED equals "what you see at the editor"
ACTUAL equals "what public_map shows"

INITIAL
3 polygon
2 bigpoints
1 smallpoints

> I move polygon layer to bottom

EXPECTED & ACTUAL
3 bigpoints
2 smallpoints
1 polygon

> I move polygon again topmost

EXPECTED:
3 polygon
2 bigpoints
1 smallpoints

ACTUAL:
3 polygon
2 smallpoints
1 bigpoints

Editor displays EXPECTED order, but upon reload is also at ACTUAL state

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
Fixes #4408.

cc/ @xavijam

### CODE REVIEW

File: lib/assets/javascripts/cartodb/table/actions_menu.js [hunk 1]
```diff
@@ -566,17 +566,19 @@
       // is a base layer at position 0
       var newIndex = $('.layer_panel').length - $('.layer_panel').index(ui.item);
       var layerId = ui.item.attr('model-id');
-      var layers = this.map.layers;
-      var layer = layers.get(layerId);
-      layers.remove(layer);
-      layers.add(layer, { at: newIndex });
```
Comment (alonsogarciapablo) [inline lib/assets/javascripts/cartodb/table/actions_menu.js]: This was adding the layer at the given index but the `order` hadn't changed so the collection was not really changed (it's automagically sorted by order).

File: lib/assets/javascripts/cartodb/table/actions_menu.js [hunk 2]
```diff
@@ -566,17 +566,19 @@
...
       // and save the new layers order to the server
-      layers.unbind('reset', this._resetLayers, this);
-      layers.saveLayers({
+      this.map.layers.unbind('reset', this._resetLayers, this);
+      this.map.layers.saveLayers({
         // TODO: what happens when there is an error?
         complete: function() {
+
+          // TODO: Move this trigger to saveLayers?
+          self.map.layers.trigger('reset');
```
Comment (alonsogarciapablo) [inline lib/assets/javascripts/cartodb/table/actions_menu.js]: I will move this when we fix the `saveLayers` method.

Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/actions_menu.js]: :+1:

**Comment by xavijam** [comment]
If the first layer of a visualization is removed, this new method will be invoked too, right?

**Comment by xavijam** [comment]
retest this please

**Comment by alonsogarciapablo** [comment]
This method is only invoked when layers are reordered through the UI. When layers are added or removed from the layers collection, order is re-assigned [here](https://github.com/CartoDB/cartodb.js/blob/develop/src/geo/map.js#L232-L233).

**Comment by xavijam** [comment]
:+1: perfect

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5053/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5058/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5062/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5063/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5064/))

**Pull request merged**